p2p/protocols/error.go

```diff
@@ -0,0 +1,22 @@
+package protocols
+
+// HandlerError wraps standard error
+// This error is handled specially by protocol.Run
+// It causes the protocol to return with ErrHandler(err)
+type breakError struct {
+	err error
+}
+
+func BreakError(err error) *breakError {
```

Comment:
Do not return unexported type. Use interface or export breakError.

Constructor should be called `NewBreakError`. Or maybe just `Break`?

Reply:
Sounds good. Hm, I was also wondering how to call it. I think `Break` sounds good to me event, I was not sure if something like this has to little info.